gen/builders/generator.go

```diff
@@ -42,9 +44,36 @@ type Generator struct {
 // genData is the generation data to stamp into vectors.
 // TODO in the future this should contain the commit of this tool and
 //  the builder api.
-var genData = GenerationData{
-	Source:  "script",
-	Version: "v0",
+var genData = []GenerationData{
+	{
+		Source:  "script",
+		Version: "v0",
+	},
+}
+
+func init() {
+	genData = append(genData, getBuildInfo()...)
+}
+
+func getBuildInfo() []GenerationData {
+	deps := []string{"github.com/filecoin-project/lotus", "github.com/filecoin-project/specs-actors"}
+
+	bi, ok := debug.ReadBuildInfo()
+	if !ok {
+		panic("cant read build info")
+	}
+
+	var result []GenerationData
+
+	for _, v := range bi.Deps {
+		for _, dep := range deps {
+			if strings.HasPrefix(v.Path, dep) {
```

Comment:
are there sub-modules within lotus with their own go.mod that will also end up in here with this?

Reply:
At the moment - no.